activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -589,25 +589,42 @@ def exec_cache(sql, name, binds)
             @connection.exec_prepared(stmt_key, type_casted_binds)
           end
         rescue ActiveRecord::StatementInvalid => e
-          pgerror = e.cause
 
-          # Get the PG code for the failure.  Annoyingly, the code for
-          # prepared statements whose return value may have changed is
-          # FEATURE_NOT_SUPPORTED.  Check here for more details:
-          # http://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/backend/utils/cache/plancache.c#l573
-          begin
-            code = pgerror.result.result_error_field(PGresult::PG_DIAG_SQLSTATE)
-          rescue
-            raise e
-          end
-          if FEATURE_NOT_SUPPORTED == code
+          raise unless is_cached_plan_failure?(e)
```

Comment:
Out of curiosity, would this be ```raise e```? 

I'm tracking this PR for our own issues with prepared statements in Rails 4-2. This is really awesome work.